File: 0001-two-sum/0001-two-sum.cpp
#include <vector>
#include <algorithm>

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> store;
        for (int i = 0; i < nums.size(); ++i) {
            int diff = target - nums[i];
            if (store.find(diff) != store.end()) {
                return {store[diff], i};
            }
            store[nums[i]] = i;
        }
        // If no solution found, return an empty vector
        return {};
    }
};

File: 0001-two-sum/0001-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    const newMap = {}
    for(let i =0; i<nums.length; i++){
        const diff = target - nums[i]
        if(diff in newMap) return [i, newMap[diff]]
        newMap[nums[i]] = i
    }
};